src/Validator/Constraints/P0wnedPasswordValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the RollerworksPasswordStrengthValidator package.
+ *
+ * (c) Sebastiaan Stok <s.stok@rollerscapes.net>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Rollerworks\Component\PasswordStrength\Validator\Constraints;
+
+use Rollerworks\Component\PasswordStrength\P0wnedPassword\Request\Client;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+class P0wnedPasswordValidator extends ConstraintValidator
+{
+    /** @var Client */
+    private $client;
+
+    /**
+     * P0wnedPasswordValidator constructor.
+     *
+     * @param Client $client
+     */
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!empty($value)) {
```

Comment:
Better to use an early return and check for a null values explicitly:

```php
        if (null === $password) {
            return;
        }

        if (!is_scalar($password) && !(is_object($password) && method_exists($password, '__toString'))) {
            throw new UnexpectedTypeException($password, 'string');
        }

        $password = (string) $password;
```

FYI. By convention all Password validators use `$password` as value argument.